pxr/usd/usdUtils/validators.cpp

```diff
@@ -0,0 +1,88 @@
+//
+// Copyright 2024 Pixar
+//
+// Licensed under the terms set forth in the LICENSE.txt file available at
+// https://openusd.org/license.
+//
+
+#include <pxr/usd/sdf/fileFormat.h>
+#include "pxr/usd/usd/validationError.h"
+#include "pxr/usd/usd/validationRegistry.h"
+#include "pxr/usd/usdUtils/validatorTokens.h"
+#include "pxr/usd/ar/packageUtils.h"
+#include "pxr/usd/usdUtils/dependencies.h"
+#include "pxr/usd/usdUtils/userProcessingFunc.h"
+
+PXR_NAMESPACE_OPEN_SCOPE
+
+static
+UsdValidationErrorVector
+_PackageEncapsulationValidator(const UsdStagePtr& usdStage) {
+    UsdValidationErrorVector errors;
+
+    const SdfLayerRefPtr& rootLayer = usdStage->GetRootLayer();
+    const bool isPackage = [](const SdfLayerHandle& layer){
+        return layer->GetFileFormat()->IsPackage() || ArIsPackageRelativePath(layer->GetIdentifier());
+    }(rootLayer);
+
+    if (!isPackage){
+        return errors;
+    }
+
+    std::vector<TfRefPtr<SdfLayer>> layers;
```

Comment:
You should be able to use SdfLayerRefPtrVector, defined here: https://github.com/PixarAnimationStudios/OpenUSD/blob/release/pxr/usd/sdf/declareHandles.h#L335